### PR TITLE
PR Template cleanup

### PR DESCRIPTION
**Pull request by mirie**
Removed some sections we've been using less often to streamline the PR review process.

### CODE REVIEW

**Review by maestromac — APPROVED** [review]

**Review by rt4914 — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -55,37 +55,6 @@ _If your PR includes UI changes, please replace this line with details on how
 accessibility is impacted and tested. For more info, check out the
 [Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._
 
-## Added/updated tests?
```
Comment (Ridhwana) [inline .github/PULL_REQUEST_TEMPLATE.md:58]: @mirie should we remove this section? I feel like this section is important for OS contributors who sometimes need help writing tests? I think its also good to be able to rationalize about the cases where one is not writing tests for the feature.

**Review by Ridhwana — COMMENTED** [review]
Left a comment about removing the ##Added/updated tests.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jaw6) [reply .github/PULL_REQUEST_TEMPLATE.md:58]: I agree with @Ridhwana here, I think this section is worth keeping.

**Comment by rt4914** [comment]
@mirie A change was suggested on this PR. Feel free to update it whenever possible.

**Pull request merged**